### PR TITLE
Remove panel

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

Addresses #4, start on #3

Removes `pandas.Panel` from package.

Starts to switch code over to using xarray as main data format rather than something pandas based.

Tweaked general averaging code to reduce specificity with pandas. Started modifying `computational_form` to enable it's use in more areas.

The code is old, standards wise. Happy to sort that out as time goes on but a decent chunk of existing code may get deleted. I'd rather not make soon to be deleted code pretty. Overall plan is to reserve general comment and docstring improvements until after the transition to supporting xarray, as well as general nD binning support, is complete.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide
instructions so we can reproduce. Please also list any relevant details for
your test configuration

- Current suite of unit tests

**Test Configuration**:
* MacOS 11.5
* Python 3.8
* Any details about your local setup that are relevant

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by rstoneback** [comment]
Current implementation requires converting each DataFrame to a Dataset. Would likely be better if we could just make a Dataset in a single go. I've been having trouble with that though and wanted to get something up. Working solution > no solution.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [0.1.4] - 
+- Removed deprecated `pandas.Panel` from functions.
+
```
Comment (rstoneback) [inline CHANGELOG.md:7]: Add note about changes to computational form

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [0.1.4] -
```
Comment (aburrell) [inline CHANGELOG.md:5]: Add a placeholder for the date so it isn't forgotten.

File: pysatSeasons/_core.py [hunk 3]
```diff
@@ -1,12 +1,13 @@
 import pandas as pds
+import xarray as xr
 
 
 def computational_form(data):
     """
-    Repackages numbers, Series, or DataFrames
+    Repackages input data into xarray.Dataset
```
Comment (aburrell) [inline pysatSeasons/_core.py:7]: Fix docstring formatting

File: pysatSeasons/_core.py [hunk 4]
```diff
@@ -17,22 +18,80 @@ def computational_form(data):
...
+        output = data.to_xarray()
+    elif isinstance(data, pds.Series):
+        output = xr.Dataset()
+        output[data.name] = data.to_xarray()
+    elif isinstance(data, xr.Dataset):
+        output = data
+    elif isinstance(data, xr.DataArray):
+        output = xr.Dataset()
+        output[data.name] = data
+
```
Comment (aburrell) [inline pysatSeasons/_core.py:47]:
```suggestion
        output[data.name] = data
```

File: pysatSeasons/_core.py [hunk 5]
```diff
@@ -1,12 +1,13 @@
 import pandas as pds
+import xarray as xr
 
 
 def computational_form(data):
```
Comment (aburrell) [inline pysatSeasons/_core.py:5]: Needs a more descriptive name.  Maybe "to_xarray_dataset"?

File: pysatSeasons/_core.py [hunk 6]
```diff
@@ -17,22 +18,80 @@ def computational_form(data):
...
+        vars = [data[0].name]
+        output = xr.Dataset()
+
+        # Combine all info for each variable into a single data
+        # array.
+        for var in vars:
+            output[var] = xr.concat(data, 'pysat_binning')
+
+    elif isinstance(data[0], pds.DataFrame):
+        # Convert data to xarray
```
Comment (aburrell) [inline pysatSeasons/_core.py:70]: This is what the whole routine does, make comment more specific.

File: pysatSeasons/avg.py [hunk 7]
```diff
@@ -1,14 +1,16 @@
 # -*- coding: utf-8 -*-
-from __future__ import print_function
-from __future__ import absolute_import
 """
 Instrument independent seasonal averaging routine. Supports averaging
 1D and 2D data.
 """
+from __future__ import print_function
+from __future__ import absolute_import
```
Comment (aburrell) [inline pysatSeasons/avg.py:7]: Remove

File: pysatSeasons/avg.py [hunk 8]
```diff
@@ -23,77 +25,74 @@ def median1D(const, bin1, label1, data_label, auto_bin=True, returnData=False):
     binX: array-like
         List holding [min, max, number of bins] or array-like containing
         bin edges, where X = 1, 2
-    labelX: string
+    labelX: str
```
Comment (aburrell) [inline pysatSeasons/avg.py:28]: Remove camelCase variable names

File: pysatSeasons/avg.py [hunk 9]
```diff
@@ -23,77 +25,74 @@ def median1D(const, bin1, label1, data_label, auto_bin=True, returnData=False):
...
     if auto_bin:
         binx = np.linspace(bin1[0], bin1[1], bin1[2] + 1)
     else:
         binx = np.array(bin1)
 
-    # how many bins are used
+    # How many bins are used
     numx = len(binx) - 1
-    # how many different data products
+    # How many different data products
```
Comment (aburrell) [inline pysatSeasons/avg.py:65]:
```suggestion

    # How many different data products
```

Comment (aburrell) [reply pysatSeasons/avg.py:65]: Have a whitespace before new comments that have the same indentation level.  Makes things easier to read.

**Review by aburrell — COMMENTED** [review]
Partial review with some style things.

File: pysatSeasons/tests/test_avg.py [hunk 10]
```diff
@@ -79,8 +79,10 @@ def test_basic_orbit_mean(self):
                                          orbit_info=orbit_info)
         self.testInst.bounds = self.bounds2
         ans = avg.mean_by_orbit(self.testInst, 'mlt')
-        # note last orbit is incomplete thus not expected to satisfy relation
-        assert np.allclose(ans[:-1], np.ones(len(ans) - 1) * 12.0, 1.0E-2)
+        # Note last orbit is incomplete thus not expected to satisfy relation
```
Comment (rstoneback) [inline pysatSeasons/tests/test_avg.py:82]:
```suggestion

        # Note last orbit is incomplete thus not expected to satisfy relation
```

**Comment by rstoneback** [comment]
Thanks for the comments @aburrell. Updated.

**Review by jklenzing — COMMENTED** [review]
Looking good.  Note that the version caps in setup.cfg need to be removed as well for when people are installing through pip.

**Review by jklenzing — APPROVED** [review]
Looks good!  Thanks @rstoneback

**Comment by jklenzing** [comment]
Just realized the changelog should probably show this as 0.2.0.

**Pull request merged**